### PR TITLE
better workload auto-injection label/annotation processing

### PR DESCRIPTION
**Pull request by jmazzitelli**
fixes: https://github.com/kiali/kiali/issues/5713

### CODE REVIEW

**Comment by jmazzitelli** [comment]
To test this, deploy an application (e.g. bookinfo). Add the `sidecar.istio.io/inject` annotation to a deployment (productpage or something like that) so the pod gets the annotation added to it. Then navigate to that Workload in Kiali and select the Action dropdown option "Disable Injection" and see that the annotation is now removed from the pod and label will be set to "false".

You can also confirm the same thing happens (the annotation is removed) when enabling injection.

This should work in both upstream Istio and OSSM environments.

**Review by jshaughn — APPROVED** [review]
Tested and seems to be working, the annotation is wiped from the deployment's template and therefore the new pod is not annotated, but labels exist.

Approving, although two small golang tweaks if you care to accept the changes.

File: models/workload.go [hunk 1]
```diff
@@ -176,24 +177,29 @@ func (workload *Workload) parseObjectMeta(meta *meta_v1.ObjectMeta, tplMeta *met
...
+			labelExplicitlySet = true
 		}
 	}
 
-	// do not bother to check the annotation if the label is explicitly false since we know in that case injection is disabled
-	if !labelExplicitlyFalse {
+	// do not bother to check the annotation if the label is explicitly set - label always overrides the annotation
+	if !labelExplicitlySet {
 		annotation, exist := annotations[conf.ExternalServices.Istio.IstioInjectionAnnotation]
 		if exist {
```
[the remark this answers is not in the document]
Comment (jmazzitelli) [reply models/workload.go:196]: this is not any code I changed in this PR, leaving as-is to avoid unnecessary churn.

**Comment by jmazzitelli** [comment]
> do we want to add test coverage for this specific scenario?

It would not hurt to have test coverage for this. The test would have to be able to get the actual yaml of the Pod itself and see that its label and annotation change.

**Pull request merged**